File: hanoi/python_hanoi/hanoi.py
import sys

def hanoi(ndisks, startPeg=1, endPeg=3):
  if ndisks:
    hanoi(ndisks-1, startPeg, 6-startPeg-endPeg)
    print( "Move disk %d from peg %d to peg %d" % (ndisks, startPeg, endPeg))
    hanoi(ndisks-1, 6-startPeg-endPeg, endPeg)


def main():
  if len(sys.argv) != 1:
    print( "Usage: %s <noargs>" % sys.argv[0])
    sys.exit()
  hanoi(ndisks=3)
  sys.exit(), end

main()

File: hanoi/ocaml_hanoi/hanoi.ml
(* move n discs from a to z (using third peg named x) *)
let rec hanoi move n a z x = if n = 0 then () else
  (hanoi move (n-1) a x z ; move n a z ; hanoi move (n-1) x z a)
let printmove disc a b = print_endline
("Move disc " ^ (string_of_int disc) ^ " from " ^ a ^ " to " ^ b) ;;
hanoi printmove 4 "peg A" "peg C" "peg B"

let main () =
  if  Array.length Sys.argv <> 1 then
    begin
      Printf.printf "%s <noargs>\n" Sys.argv.(0);
      exit 1
    end
  else
    begin
      for idx_i = 1 to 10 do
        (*Printf.printf "calculateEasterSunday(%d)=<%s>\n" idx_i (calculateEasterSunday(idx_i)) *)
      done;
    end
let _ = main ()
